### ISSUE TITLE
cecil 0.10 appears to affect the pdb such that application doesnt get source file and line number in exception stack trace

### ISSUE THREAD

**Issue by parthopdas** [jbevain/cecil#422]
**TL;DR** This is a case where the debug information being emitted by cecil 0.10 seems unreadable by the application.

Couple of caveats
1. This is a CLR 2.0 project (cecil 0.10 does say it doesn't support .net 2.0 but since it supports .net 3.5 I am assuming CLR 2.0 should be OK)
2. There is another component in the picture where the root cause could be - xunit. However since only cecil changed from 0.9.x to 0.10, I am filing the issue with cecil. (everything else in the project stayed the same including xunit versions)

The issue:
1. Please see the repro steps below where I have run the scenario without the instrumented assembly (working case) and then with the instrumented assembly (notworking case)
2. In both cases, I invoke the code (its a unit test) once through a simple console application and once through xunit console runner
3. The xunit console runner doesn't spit out the source file line number with the instrumented assembly (see the lines ending with "[FAIL]"), it does so with the un-instrumented assembly
4. In both working and notworking cases, the console application spits out the source file and line number in the exception (this could be a data point for your investigations - meaning xunit also has something to do with this issue, but the breaking change is definitely in cecil, without which the scenario would work fine)

Here are the repro steps.
1. Unzip http://s000.tinyupload.com/index.php?file_id=30492199408931230200
2. cd notworking
3. ConsoleApplication5.exe
```
   at CSXUnit1xNUnit3x.StringTests3.TestToSkip() in D:\src\t\Engine\build\f37b80a8-81e9-474c-969b-7fb815384945\CSXUnit1xNUnit3x.NET20\Class1.cs:line 42
   at ConsoleApplication5.Program.Main(String[] args) in D:\projects\ConsoleApplication5\Program.cs:line 12
```
4. xunit.console.exe CSXUnit1xNUnit3x.dll
```
xUnit.net Console Runner (64-bit .NET 4.0.30319.42000)
  Discovering: CSXUnit1xNUnit3x
  Discovered:  CSXUnit1xNUnit3x
  Starting:    CSXUnit1xNUnit3x
    CSXUnit1xNUnit3x.StringTests3.IndexOf(input: "hello world", letter: 'w', expected: 7) [FAIL]
      Assert.Equal() Failure
      Expected: 7
      Actual:   6
      Stack Trace:
           at CSXUnit1xNUnit3x.StringTests3.IndexOf(String input, Char letter, Int32 expected)
    CSXUnit1xNUnit3x.StringTests3.TestToSkip [FAIL]
      Assert.Equal() Failure
      Expected: 1
      Actual:   0
      Stack Trace:
           at CSXUnit1xNUnit3x.StringTests3.TestToSkip()
  Finished:    CSXUnit1xNUnit3x
=== TEST EXECUTION SUMMARY ===
   CSXUnit1xNUnit3x  Total: 3, Errors: 0, Failed: 2, Skipped: 0, Time: 0.138s
```

5. cd ..\working
6. ConsoleApplication5.exe
```
   at CSXUnit1xNUnit3x.StringTests3.TestToSkip() in D:\src\t\Engine\build\f37b80a8-81e9-474c-969b-7fb815384945\CSXUnit1xNUnit3x.NET20\Class1.cs:line 42
   at ConsoleApplication5.Program.Main(String[] args) in D:\projects\ConsoleApplication5\Program.cs:line 12
```
7. xunit.console.exe CSXUnit1xNUnit3x.dll
```
xUnit.net Console Runner (64-bit .NET 4.0.30319.42000)
  Discovering: CSXUnit1xNUnit3x
  Discovered:  CSXUnit1xNUnit3x
  Starting:    CSXUnit1xNUnit3x
    CSXUnit1xNUnit3x.StringTests3.TestToSkip [FAIL]
      Assert.Equal() Failure
      Expected: 1
      Actual:   0
      Stack Trace:
        D:\src\t\Engine\build\f37b80a8-81e9-474c-969b-7fb815384945\CSXUnit1xNUnit3x.NET20\Class1.cs(42,0): at CSXUnit1xNUnit3x.StringTests3.TestToSkip()
    CSXUnit1xNUnit3x.StringTests3.IndexOf(input: "hello world", letter: 'w', expected: 7) [FAIL]
      Assert.Equal() Failure
      Expected: 7
      Actual:   6
      Stack Trace:
        D:\src\t\Engine\build\f37b80a8-81e9-474c-969b-7fb815384945\CSXUnit1xNUnit3x.NET20\Class1.cs(14,0): at CSXUnit1xNUnit3x.StringTests3.IndexOf(String input, Char letter, Int32 expected)
  Finished:    CSXUnit1xNUnit3x
=== TEST EXECUTION SUMMARY ===
   CSXUnit1xNUnit3x  Total: 3, Errors: 0, Failed: 2, Skipped: 0, Time: 0.140s
   ```

**Comment by parthopdas** [comment]
This may not even be an issue. I'll close it for now with the following information.

Some additional information:
1. The xunit code does do any processing of exception. It invokes through reflection and catch the targetinvocationexception and then processes innerexception.
2. I verified the scenarios with the asmDef.Write overload that overwrites (i.e ReadWrite = true in ReadParameters) and the issues i mention above don't occur.

The above lead me to believe cecil has nothing to do with this issue. Some entity in the picture reads the pdb path embedded in the dll and fails to load the pdb from there. This may be because I have moved around the files names after instrumentation.

**Comment by parthopdas** [comment]
Correction - this issue has been introduced with the change from cecil 0.9.x to 0.10.x. Hence it definitely has something to do with cecil 0.10.x. My best guess is the pdb path naming logic has changed in 0.10.x.

However the above scenarios are work fine with the changes.
